CMakeLists.txt

```diff
@@ -0,0 +1,199 @@
+
+# Include the other modules
+include(CMakePackageConfigHelpers)
+include(DetectCompiler)
+include(DetectOS)
+include(GNUInstallDirs)
+set(PCAPP_INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR})
+set(PCAPP_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/pcapplusplus)
+set(PCAPP_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/pcapplusplus)
+
+# Build options (Turn on Examples and Tests if it's the main project)
+option(PCAPPP_BUILD_EXAMPLES "Build Examples" ${PCAPPP_MAIN_PROJECT})
+option(PCAPPP_BUILD_TESTS "Build Tests" ${PCAPPP_MAIN_PROJECT})
+option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
+
+option(PCAPPP_USE_DPDK "Setup PcapPlusPlus with DPDK. In this case you must also set DPDK_HOME")
+option(PCAPPP_USE_PF_RING "Setup PcapPlusPlus with PF_RING. In this case you must also set PF_RING_HOME")
+option(PCAPPP_INSTALL "Install Pcap++" ${PCAPPP_MAIN_PROJECT})
+
+# Set C++11
+set(CMAKE_CXX_STANDARD 11)
+# popen()/pclose() are not C++ standards
+set(CMAKE_CXX_EXTENSIONS ON)
+
+# Usually on Windows PCAP_ROOT and Packet_ROOT are at the same location
+if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+  if(NOT PCAP_ROOT)
+    message(FATAL_ERROR "Please specify Npcap/WinPcap SDK directory with -DPCAP_ROOT=<PCAP_SDK_PATH>")
+  endif()
+  if(NOT Packet_ROOT)
+    set(Packet_ROOT ${PCAP_ROOT})
+  endif()
+endif()
+
+# Do we need Packet Packages on Windows ?
+if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+  find_package(Packet)
+  if(NOT Packet_FOUND)
+    message(FATAL_ERROR "Please specify Packet library -DPacket_ROOT=<NPCAP_SDK_PATH>")
+  endif()
+endif()
+
+find_package(PCAP)
+if(NOT PCAP_FOUND)
+  if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+    message(FATAL_ERROR "Please specify Npcap/WinPcap SDK directory with -DPCAP_ROOT=<PCAP_SDK_PATH>")
+  else()
+    message(FATAL_ERROR "PCAP library not found!")
+  endif()
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_IMMEDIATE_MODE "Enable PCAP immediate Mode (supported on libpcap>=1.5)"
+       ${HAVE_PCAP_IMMEDIATE_MODE})
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_IMMEDIATE_MODE)
+  if(NOT HAVE_PCAP_IMMEDIATE_MODE)
+    message(FATAL_ERROR "PCAP library doesn't have Immediate Mode support!")
+  endif()
+  add_definitions(-DHAS_PCAP_IMMEDIATE_MODE)
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_SET_DIRECTION "Enable PCAP Direction catpure" ${HAVE_PCAP_DIRECTION})
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_SET_DIRECTION)
+  if(NOT HAVE_PCAP_DIRECTION)
+    message(FATAL_ERROR "PCAP library doesn't have Direction support!")
+  endif()
+  add_definitions(-DHAS_SET_DIRECTION_ENABLED)
+endif()
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+if(PCAPPP_TARGET_PLATFORM_WINDOWS AND PCAPPP_TARGET_COMPILER_MSVC)
+  # Other hacks for VStudio
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else()
+  if(NOT CMAKE_USE_PTHREADS_INIT)
+    message(FATAL_ERROR "Pthreads not found!")
+  endif()
+endif()
```

Comment:
Since we moved to C++11 we don't use `pthread` anymore, is this section needed?

Reply:
we still use pthread_setaffinity_np()

Reply:
But it's only used in DPDK and PF_RING which are only available on Linux, where pthreads are installed by default 🤔

Reply:
Yes but 'im not sure if multiple threads lib are installed on the system. would cmake link with the correct one ?

Reply:
this ensure that we use pthread and threads in the libstdc++ or another threads library..

Reply:
Ok, I trust you on this 👍